src/sql/workbench/contrib/notebook/browser/models/modelInterfaces.ts

```diff
@@ -413,6 +417,85 @@ export interface INotebookModel {
 
 	standardKernels: IStandardKernelWithProvider[];
 
+	getFindCount(): number;
+
+	getFindIndex(): number;
+
+	findNext(): Thenable<NotebookRange>;
+
+	findPrevious(): Thenable<NotebookRange>;
+
+	find(exp: string, maxMatches?: number): Promise<NotebookRange>;
+
+	clearFind(): void;
+
+	findArray: NotebookRange[];
+
+	/**
+	 * Get the range associated with a decoration.
+	 * @param id The decoration id.
+	 * @return The decoration range or null if the decoration was not found.
+	 */
+	getDecorationRange(id: string): NotebookRange | null;
+
+	changeDecorations<T>(callback: (changeAccessor: IModelDecorationsChangeAccessor) => T, ownerId: number): T | null;
+	/**
+	 * Get the maximum legal column for line at `lineNumber`
+	 */
+	getLineMaxColumn(lineNumber: number): number;
+
+	/**
+	 * Get the number of lines in the model.
+	 */
+	getLineCount(): number;
+
+	findMatches: NotebookFindMatch[];
```

Comment:
I haven't gotten to the logic for this yet - but I'm confused why these would be separate properties on the model. Shouldn't find be a single operation that returns the matches directly? The model shouldn't be storing the results itself I would think